lsp-mode.el

```diff
@@ -3992,6 +3998,20 @@ in that particular folder."
            (lsp--on-change-debounce buffer)
            (lsp--on-idle buffer)))))))
 
+(defun lsp-unconfig-buffer ()
+  (when lsp-modeline-code-actions-mode
+    (lsp-modeline-code-actions-mode -1))
+  (when lsp-headerline-breadcrumb-mode
+    (lsp-headerline-breadcrumb-mode -1))
+  (when (bound-and-true-p dap-auto-configure-mode)
```

Comment:
remove the dap part - otherwise disconnecting from one buffer will turn off dap-mode globally.